### PR TITLE
cert-manager delegation stage 2

### PR DESCRIPTION
**Pull request by qpdpQ**

### CODE REVIEW

File: controllers/cert-manager/podrefresh_controller.go [hunk 1]
```diff
@@ -323,8 +327,55 @@ func (r *PodRefreshReconciler) updateDaemonSetAnnotations(daemonsetsToUpdate []a
...
+			}
+		}
+	}
+	return false, nil
+}
+
 // SetupWithManager sets up the controller with the Manager.
 func (r *PodRefreshReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// wait for crd ready
+	if err := r.waitResourceReady("apiextensions.k8s.io/v1", "CustomResourceDefinition"); err != nil {
```
Comment (bitscuit) [inline controllers/cert-manager/podrefresh_controller.go:375]:
```suggestion
	if err := r.waitResourceReady("cert-manager.io/v1", "Certificate"); err != nil {
```

Shouldn't we be waiting for the cert-manager CRD specifically?

File: controllers/certmanager/certificate_controller.go [hunk 2]
```diff
@@ -435,8 +439,55 @@ func (r *CertificateReconciler) updateLeafCerts(issuers []certmanagerv1alpha1.Is
...
+			}
+		}
+	}
+	return false, nil
+}
+
 // SetupWithManager sets up the controller with the Manager.
 func (r *CertificateReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// wait for crd ready
+	if err := r.waitResourceReady("apiextensions.k8s.io/v1", "CustomResourceDefinition"); err != nil {
```
Comment (bitscuit) [inline controllers/certmanager/certificate_controller.go:487]: Again, shouldn't we be watching for cert-manager CRD (either Issuer or Certificate depending on the controller) specifically?

File: controllers/operator/v1CRDs.go [hunk 3]
```diff
@@ -0,0 +1,4314 @@
...
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: certificaterequests.cert-manager.io
+  labels:
+    app: 'cert-manager'
+    app.kubernetes.io/name: 'cert-manager'
+    app.kubernetes.io/instance: 'cert-manager'
+    # Generated labels
+    app.kubernetes.io/version: "v1.9.0-alpha.0"
```
Comment (bitscuit) [inline controllers/operator/v1CRDs.go:29]: We should use the v1.7.1 CRDs because that is what we deploy

**Review by bitscuit — CHANGES_REQUESTED** [review]

**Review by Daniel-Fan — APPROVED** [review]
/lgtm

**Pull request merged**